oauth2/src/main/java/com/linecorp/armeria/client/auth/oauth2/OAuth2AuthorizationGrant.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.auth.oauth2;
+
+import java.util.concurrent.CompletionStage;
+
+import com.linecorp.armeria.common.auth.oauth2.AccessTokenCapsule;
+
+/**
+ * Represents an OAuth 2.0 Access Token Grant flow to obtain Access Token.
+ */
+@FunctionalInterface
+public interface OAuth2AuthorizationGrant {
+
+    /**
+     * Produces OAuth 2.0 Access Token
+     */
+    CompletionStage<AccessTokenCapsule> getAccessToken();
```

Comment:
Is the term `Capsule` something defined in `OAuth 2` or an authorization jargon? If not, what would be a better name? `GrantedAccessToken`? :thinking:

Reply:
No, `Capsule` is not part of `OAuth 2`. It's just a word to describe a shell or a container that contains `Access Token`, `Refresh Token` and some other elements. It's not really a descriptor (which is what we get as a result of introspection). You don't like the word `Capsule` or you think it could be confusing?

Reply:
I prefer `OAuth2AccessToken` 😉

Reply:
Renamed to `OAuth2AccessToken`